File: DeepSpace/src/deepspace/WeaponType.java
package deepspace;

/**
 * @author miguelangelmc
 */
public enum WeaponType {
    LASER(2.0),
    MISSILE(3.0),
    PLASMA(4.0);
    
    private final double power;
    
    WeaponType(double power){
        this.power = power;
    }
    
    public double getPower(){ return power; }
}

File: deepspace/WeaponType.rb
module WeaponType
  class Type
    def initialize(value)
      @power = value
    end

    def getPower
      @power
    end
  end

  LASER = Type.new(2.0)
  MISSILE = Type.new(3.0)
  PLASMA = Type.new(4.0)
end
